src/Pages/ProductPage/ProductPage.tsx

```diff
@@ -0,0 +1,82 @@
+import { FC, useEffect, useState } from "react";
+import { Content } from "antd/es/layout/layout";
+import { ShoppingCartOutlined } from "@ant-design/icons";
+import { Spin, Skeleton, Button, message } from "antd";
+import { useParams, useNavigate, Link } from "react-router-dom";
+import { getGoods, getGoodsLoadStatus, goodActions, cartActions, getCartGoods, getIsAuthValue } from "src/store";
+import { useSelector } from "react-redux";
+import { useAppDispatch } from "src/hooks/useAppDispatch";
+import { Good } from "src/types/general";
+import { api } from "src/api/api";
+import css from "./productPage.module.css";
+
+export const ProductPage: FC = () => {
+    const loadStatus = useSelector(getGoodsLoadStatus);
+    const isAuth = useSelector(getIsAuthValue)
+    const navigate = useNavigate();
+    const { ids } = useParams();
+    const dispatch = useAppDispatch();
+    const goods = useSelector(getGoods);
+    const cartGoods = useSelector(getCartGoods);
+    const [count, setCount] = useState<number>(1);
+    const [currentGoods, setCurrentGoods] = useState<Good | null>();
```

Comment:
Если что-то можно рассчитать из уже имеющихся данных, то мы должны это рассчитывать, а не дублировать стейт. Дублированный стейт - дорога к избыточному коду и багам (т.к. всегда надо синхронизировать дублирующиеся данные). В данном случае данные у тебя лежат в goods. Обращайся к нему вместо создания нового состояния
```suggestion
    const currentGoods: Good | null = goods[0] ?? null;
```